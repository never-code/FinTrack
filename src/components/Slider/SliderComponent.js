import React from 'react';
import './Slider.scss';
import { formatCurrency } from '../../utils/helper';

const SliderComponent = ({ value, min, max, step, onChange, onBlur, onFocus, unit, name }) => (
  <div className="slider-container">
    {/* <div className="popup-value">
      <span className="popup">{value}</span>
    </div> */}
    <div className="selected-value">
      <span className="value">{name === 'loanAmount' ? formatCurrency(value * 10000, 0) : value}</span>
      {unit && <span className="unit">{unit}</span>}
    </div>
    <div className="field">
      <div className="value min">{min}</div>
      <input
        type="range"
        min={min}
        max={max}
        defaultValue={value}
        step={step}
        onChange={onChange}
      />
      <div className="value max">{max}</div>
    </div>
  </div>
);

export default SliderComponent;
