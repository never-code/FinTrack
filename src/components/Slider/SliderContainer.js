import React, { useState } from 'react';
import SliderComponent from './SliderComponent';

const SliderContainer = ({ min, max, step, unit, getValue, defaultValue, name }) => {
  const [state, setState] = useState({ value: defaultValue });
  const onChange = ({ target: { value } }) => {
    setState({
      ...state,
      value,
    });
    getValue(value);
  };
  const { value } = state;
  return (
    <SliderComponent
      max={max}
      min={min}
      step={step}
      value={value}
      onChange={onChange}
      unit={unit}
      name={name}
    />
  );
};

export default SliderContainer;
