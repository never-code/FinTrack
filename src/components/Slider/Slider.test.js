import React from 'react';
import { render } from '@testing-library/react';
import SliderContainer from './SliderContainer';
import SliderComponent from './SliderComponent';

describe('SliderContainer ', () => {
  it('should render successfully', () => {
    const { container } = render(<SliderContainer />);
    expect(container).toMatchSnapshot();
  });
});

describe('SliderComponent ', () => {
  it('should render successfully', () => {
    const { container } = render(<SliderComponent />);
    expect(container).toMatchSnapshot();
  });
});
