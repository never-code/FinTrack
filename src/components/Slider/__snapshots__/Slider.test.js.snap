// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderComponent  should render successfully 1`] = `
<div>
  <div
    class="slider-container"
  >
    <div
      class="selected-value"
    >
      <span
        class="value"
      />
    </div>
    <div
      class="field"
    >
      <div
        class="value min"
      />
      <input
        type="range"
        value=""
      />
      <div
        class="value max"
      />
    </div>
  </div>
</div>
`;

exports[`SliderContainer  should render successfully 1`] = `
<div>
  <div
    class="slider-container"
  >
    <div
      class="selected-value"
    >
      <span
        class="value"
      />
    </div>
    <div
      class="field"
    >
      <div
        class="value min"
      />
      <input
        type="range"
        value=""
      />
      <div
        class="value max"
      />
    </div>
  </div>
</div>
`;
